libraries/ui/src/main/java/androidx/media3/ui/PlayerControlView.java

```diff
@@ -1574,26 +1574,31 @@ private void seekToTimeBarPosition(Player player, long positionMs) {
   }
 
   private void onFullScreenButtonClicked(View v) {
-    updateIsFullScreen(!isFullScreen);
+    updateIsFullscreen(!isFullScreen);
   }
 
   /**
-   * Updates whether the controller is in full screen to its icon
-   * and reports it to to the listener.
-   * @param isFullScreen If the view is in full screen.
+   * Updates whether the controller is in fullscreen, changing its fullscreen icon and reports it
+   * to to the listener.
+   *
+   * <p>For {@code isFullscreen} equals {@code true} the icon will be set
+   * to {@link R.drawable#exo_styled_controls_fullscreen_exit} or else
+   * {@link R.drawable#exo_styled_controls_fullscreen_enter}.
+   *
+   * @param isFullscreen If the view is in full screen.
    */
-  public void updateIsFullScreen(boolean isFullScreen) {
-    boolean fullScreenModeChanged = this.isFullScreen != isFullScreen;
+  public void updateIsFullscreen(boolean isFullscreen) {
+    boolean fullscreenModeChanged = this.isFullScreen != isFullscreen;
 
-    if(fullScreenModeChanged && onFullScreenModeChangedListener == null) {
+    if (fullscreenModeChanged && onFullScreenModeChangedListener == null) {
```

Comment:
@theskyblockman, please see mine and icbaker's convo above:

tldr; 
* `if (!fullScreenModeChanged)` to exit quickly
* whether or not we have the listener is not important at this point
* If the mode DID change, we still want to update the state and go through the rest of this function

Reply:
Changed in 616e17af.